File: Core/value/stringvalue.cpp
#include "stringvalue.h"

StringValue::StringValue() : ValueBase() {
}

StringValue::StringValue(ValueGroup *valueGroup, QString id, VALUE_TYPE valueType, bool alwaysEmit, QObject *parent) : ValueBase(valueGroup, id, valueType, alwaysEmit, QVariant::String, parent)
{

}

QVariant StringValue::_updateValue(QVariant newValue) {
    if (newValue.canConvert(QVariant::String)) {
        return QVariant::fromValue(newValue);
    } else {
        return QVariant();
    }
}

File: AndroidApp/OshCore/src/main/java/com/osh/value/StringValue.java
package com.osh.value;

import com.osh.SerializationSupport;

public class StringValue extends ValueBase<StringValue, String> implements SerializationSupport {

	@Override
	protected String _updateValue(Object newValue) {
		if (newValue instanceof String) {
			return (String) newValue;
		} else if (newValue instanceof Number) {
			return ((Number) newValue).toString();
		} else if (newValue instanceof Boolean) {
			return ((Boolean) newValue).toString();
		} else {
			return null;
		}
	}

	public StringValue(ValueGroup valueGroup, String id, ValueType valueType) {
		super(valueGroup, id, valueType);
	}

	public StringValue() {
	}

}
